cdm/src/main/java/ucar/nc2/ft2/coverage/simpgeometry/SimpleGeometryCoverage.java

```diff
@@ -0,0 +1,228 @@
+ * See LICENSE for license information.
+ */
+package ucar.nc2.ft2.coverage.simpgeometry;
+
+import ucar.ma2.DataType;
+import ucar.ma2.InvalidRangeException;
+import ucar.ma2.IsMissingEvaluator;
+import ucar.ma2.Section;
+import ucar.nc2.Attribute;
+import ucar.nc2.AttributeContainerHelper;
+import ucar.nc2.Dimension;
+import ucar.nc2.VariableSimpleIF;
+import ucar.nc2.dataset.CoordinateAxis;
+import ucar.nc2.util.Indent;
+import ucar.nc2.ft2.coverage.simpgeometry.CFGEOMETRY;
+import ucar.nc2.ft2.coverage.CoverageReader;
+import ucar.nc2.ft2.coverage.SubsetParams;
+import ucar.nc2.ft2.coverage.adapter.SimpleGeometryCS;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Formatter;
+import java.util.List;
+
+/**
+ * SimpleGeometry - forked from Coverage.java
+ * Immutable after setCoordSys() is called.
+ *
+ * @author Katie
+ * @author Carron
+ * @since 8/13/2018
+ */
+// @Immutable
+public class SimpleGeometryCoverage implements VariableSimpleIF, IsMissingEvaluator { //add interface with geometry info? //subsetting data will be diffferent?? is it subset now
+  private final String name;
+  private final DataType dataType;
+  private final AttributeContainerHelper atts;
+  private final String units, description;
+  private final String coordSysName;
+  protected final CoverageReader reader;
+  protected final Object user;
+  
+  private final CFGEOMETRY geometry; //use enum?
+
+  private SimpleGeometryCS coordSys; // almost immutable use coordsys that winor made?
+
+  public SimpleGeometryCoverage(String name, DataType dataType, List<Attribute> atts, String coordSysName, String units, String description, CoverageReader reader, Object user, CFGEOMETRY geometry) {
+    this.name = name;
+    this.dataType = dataType;
+    this.atts = new AttributeContainerHelper(name, atts);
+    this.coordSysName = coordSysName;
+    this.units = units;
+    this.description = description;
+    this.reader = reader;
+    this.user = user;
+    this.geometry = geometry;
+  }
+
+
+  void setCoordSys (SimpleGeometryCS coordSys) {
+    if (this.coordSys != null) throw new RuntimeException("Cant change coordSys once set");
+    this.coordSys = coordSys;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+
+  @Override
+  public DataType getDataType() {
+    return dataType;
+  }
+
+  @Override
+  public List<Attribute> getAttributes() {
+    return atts.getAttributes();
+  }
+
+  @Override
+  public Attribute findAttributeIgnoreCase(String name) {
+    return atts.findAttributeIgnoreCase(name);
+  }
+
+  @Override
+  public String getUnitsString() {
+    return units;
+  }
+
+  @Override
+  public String getDescription() {
+    return description;
+  }
+
+  public String getCoordSysName() {
+    return coordSysName;
+  }
+
+  public Object getUserObject() {
+    return user;
+  }
+  public CFGEOMETRY getGeometry() {
+	  return geometry; 
+  }
+  public String getGeometryDescription() {

```

Comment:
No action needed on this, but just wanted to comment that an alternate way to implement this without a switch is to make the enum have a `description` field with a public accessor method, which would reduce the body of this method to...

```
public String getGeometryDescription() {
    return this.geometry.getDescription();
}
```

See section "Enum Fields and Methods" of this tutorial:
https://dzone.com/articles/using-java-enums

If it will be important for CFGEOMETRY descriptions to be consistent in multiple places, enum fields + methods could scale better as usage proliferates. It's up to you whether that is a relevant concern to address now.

Reply:
fixed.